backend/image_analyzer.py

```
@@ -284,17 +182,9 @@ def _with_structured_output(self, analysis_text: str) -> dict:
                 except (json.JSONDecodeError, Exception):
                     pass
             
-            return {
-                "success": False,
-                "error": f"Could not parse structured output: {str(e)}",
-                "raw_text": analysis_text
-            }
+            raise Exception(f"Could not parse structured output: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)